docs/src/manual/performance_tips.md

```diff
@@ -0,0 +1,47 @@
+
+If you're struggling to improve performance, feel free to ask for help in the [community forum](https://jump.dev/forum). Before asking a question, make sure to read the post [make it easier to help you](https://discourse.julialang.org/t/psa-make-it-easier-to-help-you/14757), which contains a number of tips on how to ask a good question.
+
+## Faster problem formulation: Avoid scalar indexing
+
+When manipulating Convex.jl expressions, try to use a "vectorized" style, and avoid loops and scalar indexing. For example:
+
+```julia
+function bad_my_dot(x, y) # avoid this kind of code for Convex.jl expressions!
+    s = 0.0
+    for i in eachindex(x, y)
+        s += x[i]*y[i]
+    end
+    return s
+end
+```
+
+is usually perfectly reasonable Julia code (although perhaps `s` should be initialized as `s=zero(promote_type(eltype(x), eltype(y))))` to be more generic), since loops are typically fast in Julia.
+
+However, when using Convex.jl expressions like `x = Variable(100)`, calling `bad_my_dot(x, y)` with `y = rand(100)` will create 100 separate `IndexAtom`s, each of which has some overhead. That is because Convex maintains a lazy expression tree representing the problem. Instead, the vectorized form `sum(x .* y)` or `LinearAlgebra.dot(x, y)` will be more efficient.
+
+Indexing style typically should have no effect on solve times (step (3) above), only on constructing the expression tree and formulating the problem.
+
+## Speeding up solve times: dualization
+
+Depending on the solver and the problem, it can sometimes speed things up quite a lot to pass the solver the dual problem to solve rather than the primal problem. This can be accomplished easily with [Dualization.jl](https://github.com/jump-dev/Dualization.jl), simply by passing `dual_optimizer(optimizer)` instead of `optimizer` to `solve!`. See [Dualization](@ref) for an example.
+
+## Speeding up solve times: conic-friendly formulations
+
+Convex.jl is built around conic programming, and it works with conic solvers. This may be a different optimization methodology than you may be used to, and it works by formulating problems in terms of affine objective functions and affine-function-in-cone constraints for a variety of convex cones. Convex.jl reformulates all problems to this form. For example, the objective function you give Convex will not actually be executed line-by-line; instead, it will be reformulated to a possibly very-different looking form for the conic solver to handle. This has important consequences for performance.
+
+For example, it is typically easier for conic solvers to minimize $\|A*x\|$ than $\|A*x\|^2$, since the former is reformulated as a second order cone constraint, and the latter as the same second order cone constraint, plus a rotated second order cone constraint (to model the final squaring). This may be unintuitive, since for non-conic solvers, often $\|A*x\|^2$ is easier to solve (as it avoids a square-root in the typical formulation).
```

Comment:
This is not really true, since the latter can be a rotated second order cone constraint.

Reply:
Right now we make two constraints like l wrote. Maybe there’s a better example though?

Reply:
should I just delete this section? I feel like there's a pedagogical reason to discuss this stuff but I'm not 100% sure how actionable it is for "performance tips", since Convex basically makes you formulate conic-friendly things and I haven't seen this come up that often in practice. I guess the actionable thing is: "the problem you write down is not the problem being solved, so you may need to dig in and break the abstraction if you want to top performance"

Reply:
(filed https://github.com/jump-dev/Convex.jl/issues/677 btw)

Reply:
I rewrote this section to be about "look at the final problem formulation". I believe the only stable API we have for this is `write_to_file`, so I used that.

BTW, it would probably be more standard to put the filename first (like in `Base.write`), but probably it's not worth changing from the MOI convention here.